src/osp/tasks/top_session.h

```diff
@@ -38,7 +39,13 @@
 #include <vector>
 
 #define OSP_AUX_DCDI_C(session, topData, count, ...) \
-    session.m_data.resize(count); \
+    session.m_data.resize(count, lgrn::id_null<osp::TopDataId>()); \
+    for (osp::TopDataId const id : session.m_data) \
```

Comment:
probably not worth addressing, but it's possible that the parameter `session` might be a complex expression.

you could bind to a local reference variable to avoid evaluating the expression more than once.

Reply:
I see. There are no cases of this anywhere as of now.